prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -119,9 +131,27 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
                         logger.info("Token successfully added!")
 
                         // post the report
-                        val reportClient = httpClient ?: createDefaultHttpClient(
-                            jksCredential,
-                            accessToken,
+                        // If encryption is needed.
+                        val encryptClient = httpClient ?: createDefaultHttpClient(
+                            jksCredential, accessToken,
+                            restTransportInfo
+                        )
+                        val reportContent = if (restTransportInfo.encryptionKeyUrl.isNotEmpty()) {
+                            encryptTheReport(
+                                reportContentOrg,
+                                fileName,
+                                restTransportInfo.encryptionKeyUrl,
+                                httpHeaders,
+                                logger,
+                                encryptClient
+                            )
+                        } else {
```

Comment:
Is there a risk of a misconfiguration (e.g. no encryption key being sent for the receiver) causing a security incident? How do we ensure this doesn't happen in Production?

Reply:
If there is no encryption key being sent, Natus will not be able to decrypt the message.  In fact, the encryption key is provided by them.  We retrieve the encryption key and IV from their provided GetFileEncryptionKey endpoint.  They manage it and not us.

Reply:
Ah -- okay. But if they fail to return a key for some reason then what happens? It looked like we might be sending the HL7 message unencrypted, which may trigger a security incident on their end. In this case, should we instead refuse to send them the message?

Reply:
Very good question, indeed.  No, it will not send unencrypted messages on the wire.  It will throw an exception and retry.  If retried failed, it will post the failure onto the last-mile-failure Admin UI.

Reply:
I see now. I misunderstood what "reportContentOrg" contained.